initial commit

amirahmedimtiaz opened the pull request.

archanaatr commented on material-switch/src/switchModel.ets:
```diff
@@ -0,0 +1,225 @@
+export namespace switchModel {
+  export class Model {
+    private switchId: number = 0
+    private switchSize: number = 16
```
mark all dimentions as vp

archanaatr commented on material-switch/src/switchModel.ets:
```diff
@@ -0,0 +1,225 @@
+      if (onBackgroundColor != null && onBackgroundColor!= undefined) {
+        this.onBackgroundColor = onBackgroundColor
+      } else {
+        this.onBackgroundColor = "rgba(55,136,62,1.0)"
```
default values can be defines as constant in the top and used

archanaatr commented on material-switch/src/switchModel.ets:
```diff
@@ -0,0 +1,225 @@
+
+    reset(): Model {
+      this.switchId = 0
+      this.switchSize = 16
```
all dimentions pls mark as vp

archanaatr commented on material-switch/src/switch.ets:
```diff
@@ -0,0 +1,89 @@
+
+  private onClick() {
+    if (this.onSelect != null) {
+      this.onSelect(this.model.getSwitchId(),!this.model.getIsOn());
```
format code in all ets files

archanaatr commented on material-switch/src/switch.ets:
```diff
@@ -0,0 +1,89 @@
+        duration: 100,
+        onFinish: () => {
+          if (this.model.getIsOn() == false) {
+            this.model.setSwitchSize(16)
```
all dimentions in all the codes to be marked as vp

archanaatr reviewed (commented):
please check the comments

archanaatr commented on .gitignore:
```diff
@@ -0,0 +1,5 @@
+/node_modules
```
remove .DS_Store file

archanaatr commented on README.en.md:
````diff
@@ -0,0 +1,81 @@
+
+Install using npm
+
+```npm i @ohos-material-switch```
````
update library names as per the format shared

archanaatr commented on entry/src/main/config.json:
```diff
@@ -0,0 +1,68 @@
+{
```
remove all these files
entry/src/main/.DS_Store
entry/src/main/config.json
.DS_Store

archanaatr reviewed (commented):
pls check the comments.

HMOS-Developer commented on material-switch/package.json:
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "material-switch",
```
update name to - @ohos/material-switch

HMOS-Developer commented on package.json:
```diff
@@ -0,0 +1,18 @@
+    "hypium": "^1.0.0",
+    "@ohos/hvigor": "1.0.6",
+    "@ohos/hvigor-ohos-plugin": "1.0.6",
+    "@ohos/material-switch": "file:./material-switch"
```
remove this dependency from here

HMOS-Developer commented on package.json:
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "@ohos/material-switch",
```
update name to - material-switch

HMOS-Developer commented on entry/package.json:
```diff
@@ -0,0 +1,13 @@
+  "description": "example description",
+  "repository": {},
+  "license": "ISC",
+  "dependencies": {}
```
add library dependency here - "@ohos/material-switch": "file:./material-switch"

archanaatr commented on material-switch/src/switch.ets:
```diff
@@ -0,0 +1,90 @@
+
+  private onClick() {
+    if (this.onSelect != null) {
+      this.onSelect(this.model.getSwitchId(),!this.model.getIsOn());
```
Code format to be done - Open each ets file and do Ctrl + Alt +  L

archanaatr commented on material-switch/src/switch.ets:
```diff
@@ -0,0 +1,90 @@
+            (this.model.getIsOn() ? '20vp' : '2vp')
+                                           :
+            (this.model.getIsOn() ? '23vp' : '5vp'),
+          y: this.model.getShowAnimation() ? '2vp' : ((16 - (Number(this.model.getSwitchSize()
```
if SwitchSize is 2vp, then substring will be '2v' and Number(2v) can cause abnormal exceptions, please consider such 
 cases and correct

archanaatr commented on material-switch/src/switchModel.ets:
```diff
@@ -0,0 +1,235 @@
+    private onBackgroundColor: string | Color | Resource = this.colorConstant.defaultOnBackgroundColor
+    private offBackgroundColor: string | Color | Resource = this.colorConstant.defaultOffBackgroundColor
+
+
```
remove multiple new empty lines

archanaatr reviewed (commented):
please check the comments.

amirahmedimtiaz commented:
switchSize variable was not supposed to be changed by user because code was written for specific measurements..So i have removed the  switchSize variable and modified and formatted the code
